lnd.go

```diff
@@ -713,6 +713,8 @@ func waitForWalletPassword(grpcEndpoints, restEndpoints []net.Addr,
 	// Set up a new PasswordService, which will listen for passwords
 	// provided over RPC.
 	grpcServer := grpc.NewServer(serverOpts...)
+	// This server is not used outside this function
+	defer grpcServer.GracefulStop()
```

Comment:
should we use `Stop()` instead? From the doc it says that 
```
GracefulStop stops the gRPC server gracefully. It stops the server from accepting new connections and RPCs and blocks until all the pending RPCs are finished.
```
which sounds like could lock shutdown.

Reply:
My first try was with stop, as a side effect the wallet unlock call fails in the eyes of the caller since their grpc connection is dropped mid call. graceful stop will wait until all rpc finished so the unlocker gets his response correctly. 

regarding locking, it can lock waiting for existing calls to finish. I'm not sure that's a problem in practice here since the wallet unlocker service exposes simple calls (nothing long lived like sendpayment). If you feel this is a potential issue, the defer can run a goroutine.

Reply:
Ah, yeah in that case it makes sense to do it gracefully. Would possibly be more risky to do graceful stop of the main grpc server.

Reply:
I guess it's a design choice for the main grpc server. Do you want to shutdown immediately? Or is it a good idea to wait while a payment is executing (which could possible take a while)?

AFAIK in the current state of LND in terms of continuing a payment across restarts I feel a GracefulShutdown would be best. But once the whole reliable payments family of PRs land it stops being that critical to be graceful about it.